cns/restserver/ipam.go

```diff
@@ -478,101 +639,228 @@ func (service *HTTPRestService) MarkExistingIPsAsPendingRelease(pendingIPIDs []s
 	return nil
 }
 
-func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) (cns.PodIpInfo, bool, error) {
-	var (
-		podIpInfo cns.PodIpInfo
-		isExist   bool
-	)
-
+// Returns the current IP configs for a pod if they exist
+func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) ([]cns.PodIpInfo, bool, error) {
 	service.RLock()
 	defer service.RUnlock()
 
-	ipID := service.PodIPIDByPodInterfaceKey[podInfo.Key()]
-	if ipID != "" {
-		if ipState, isExist := service.PodIPConfigState[ipID]; isExist {
-			err := service.populateIPConfigInfoUntransacted(ipState, &podIpInfo)
-			return podIpInfo, isExist, err
-		}
+	numIPConfigs := len(service.PodIPIDByPodInterfaceKey[podInfo.Key()])
+	podIPInfo := make([]cns.PodIpInfo, numIPConfigs)
+	ipConfigExists := false
 
-		logger.Errorf("Failed to get existing ipconfig. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt")
-		return podIpInfo, isExist, fmt.Errorf("Failed to get existing ipconfig. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt")
+	for i, ipID := range service.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+		if ipID != "" {
+			if ipState, isExist := service.PodIPConfigState[ipID]; isExist {
+				if err := service.populateIPConfigInfoUntransacted(ipState, &podIPInfo[i]); err != nil {
+					return podIPInfo, isExist, err
+				}
+				ipConfigExists = true
+			} else {
```

Comment:
since you return in this branch, if you flip the conditional you don't need an `else` at all

Reply:
If I flip this I won't be able to use the variables created in the if statement though outside of it right? Both `ipState` and `isExist` are used inside of that if statement and if I take those lines out wouldn't I not be able to use them?